sklearn/utils/estimator_checks.py

```diff
@@ -499,19 +498,12 @@ def check_estimator(Estimator, generate_only=False):
     if generate_only:
         return checks_generator
 
-    xfail_checks = _safe_tags(estimator, '_xfail_test')
-
     for estimator, check in checks_generator:
-        check_name = _set_check_estimator_ids(check)
-        if xfail_checks and check_name in xfail_checks:
-            # skip tests marked as a known failure and raise a warning
-            msg = xfail_checks[check_name]
-            warnings.warn(f'Skipping {check_name}: {msg}', SkipTestWarning)
-            continue
```

Comment:
All changes to `check_estimator` revert the modifications in https://github.com/scikit-learn/scikit-learn/pull/16507 since they were no longer relevant.